File: server/internal/dto/file.go
/*
项目名称：JeriBlog
文件名称：file.go
创建时间：2026-04-16 15:00:50

系统用户：Jerion
作　　者：Jerion
联系邮箱：416685476@qq.com
功能描述：文件数据传输对象
*/

package dto

import (
	"flec_blog/pkg/upload"
	"flec_blog/pkg/utils"
)

// ============ 通用文件请求 ============

// UploadFileRequest 文件上传请求
type UploadFileRequest struct {
	Type string `form:"type" binding:"required"`
}

// ============ 通用文件响应 ============

// FileUploadForWebResponse 文件上传响应
type FileUploadForWebResponse struct {
	OriginalName string `json:"original_name"`
	FileURL      string `json:"file_url"`
}

// ============ 后台文件管理请求 ============

// ListFilesRequest 文件列表请求
type ListFilesRequest struct {
	Type       string `form:"type"`
	Page       int    `form:"page,default=1" binding:"min=1"`
	PageSize   int    `form:"page_size,default=20" binding:"min=1,max=1000"`
	Keyword    string `form:"keyword"`     // 搜索关键词（文件名、原始文件名）
	Status     *int   `form:"status"`      // 状态筛选（0=未使用，1=使用中）
	UploadType string `form:"upload_type"` // 上传类型筛选
}

// ============ 后台文件管理响应 ============

// FileResponse 文件信息响应
type FileResponse struct {
	ID           uint           `json:"id"`
	OriginalName string         `json:"original_name"`
	FileName     string         `json:"file_name"`
	FileSize     int64          `json:"file_size"`
	FileType     string         `json:"file_type"`
	FileURL      string         `json:"file_url"`
	UploadType   upload.Type    `json:"upload_type"`
	UserID       *uint          `json:"user_id"`
	Status       int            `json:"status"` // 文件状态：0=未使用 1=使用中
	UploadTime   utils.JSONTime `json:"upload_time"`
}

File: admin/src/types/file.ts
/*
项目名称：JeriBlog
文件名称：file.ts
创建时间：2026-04-16 15:08:10

系统用户：Jerion
作　　者：Jerion
联系邮箱：416685476@qq.com
功能描述：类型定义 - file类型
*/

// 文件信息
export interface FileInfo {
  id: number
  file_name: string
  original_name: string
  file_url: string
  file_type: string
  file_size: number
  upload_type: string
  upload_time: string
  status: number // 0:未使用 1:使用中
}

// 文件列表查询
export interface FileListQuery {
  page?: number
  page_size?: number
  type?: string
}

// 文件查询参数
export interface FileQuery {
  page: number
  page_size: number
  keyword?: string
  status?: number
  upload_type?: string
}

// 文件列表响应
export interface FileListData {
  list: FileInfo[]
  total: number
  page: number
  page_size: number
}
